test/unit/org/apache/cassandra/cql3/validation/operations/CQLVectorTest.java

```diff
@@ -554,4 +562,97 @@ public void udf() throws Throwable
         // make sure the function referencing the UDT is dropped before dropping the UDT at cleanup
         execute("DROP FUNCTION " + f);
     }
+
+    @Test
+    public void testCopyFloatVectorFromFile() throws IOException
+    {
+        assertCopyOfVectorLiteralsFromFileSucceeds("float", 6, new Object[][] {
+            row(1, vector(0.1f, 0.2f, 0.3f, 0.4f, 0.5f, 0.6f)),
+            row(2, vector(-0.1f, -0.2f, -0.3f, -0.4f, -0.5f, -0.6f)),
+            row(3, vector(0.9f, 0.8f, 0.7f, 0.6f, 0.5f, 0.4f))
+        });
+
+        assertCopyOfVectorLiteralsFromFileSucceeds("float", 3, new Object[][] {
+            row(1, vector(0.1f, 0.2f, 0.3f)),
+            row(2, vector(-0.4f, -0.5f, -0.6f)),
+            row(3, vector(0.7f, 0.8f, 0.9f))
+        });
+    }
+
+    @Test
+    public void testCopyIntVectorFromFile() throws IOException
+    {
+        assertCopyOfVectorLiteralsFromFileSucceeds("int", 6, new Object[][] {
+            row(1, vector(1, 2, 3, 4, 5, 6)),
+            row(2, vector(-1, -2, -3, -4, -5, -6)),
+            row(3, vector(9, 8, 7, 6, 5, 4))
+        });
+
+        assertCopyOfVectorLiteralsFromFileSucceeds("int", 3, new Object[][] {
+            row(1, vector(1, 2, 3)),
+            row(2, vector(-4, -5, -6)),
+            row(3, vector(7, 8, 9))
+        });
+    }
+
+    private void assertCopyOfVectorLiteralsFromFileSucceeds(String vectorType, int vectorSize, Object[][] rows) throws IOException
```

Comment:
I was thinking myself where to put this test, and in the end, I decided to place it among the already existing CQL vector tests as the vector aspect is the common denominator for them.
When I look at it now I think the direct usage of CQLSH is enough of a reason to put it in the `org.apache.cassandra.tools.cqlsh` package. It will also allow to remove `requireNetwork` before each test case.
We could place those tests in a separate file but I think moving the creation of a test table to a `beforeClass` section is not the right motivation to do so. Especially since my test cases use tables with different vector sizes and it's more convenient to create them within the test case - less confusing for the reader as well because you don't have to scroll up and down while reading the test.
Considering all of the above I will move them to https://github.com/apache/cassandra/blob/trunk/test/unit/org/apache/cassandra/tools/cqlsh/CqlshTest.java.

Reply:
Introducing a table with two types will needlessly complicate the importing code. I don't find it justifiable, especially considering the numerous existing tests that follow a one-table-per-test-case approach (as seen in, for example, [CQLVectorTest](https://github.com/apache/cassandra/blob/3ed09031312a6b32d01ed5a647a65995206bdb75/test/unit/org/apache/cassandra/cql3/validation/operations/CQLVectorTest.java)). Therefore, I don't believe it's necessary to strive to minimize it, especially that CQLTester already provides a nice framework that abstracts away the table cleanup logic.

Reply:
Yes, and the same table is created in multiple test cases of this class which supports the claim from my previous comment.